src/site/content/en/learn/css/box-model/index.md

```diff
@@ -0,0 +1,321 @@
+---
+layout: course
+title: Box Model
+authors:
+  - andybell
+tags:
+  - css
+---
+
+Say you have this bit of HTML:
+
+```html
+<p>I am a paragraph of text that has a few words in it.</p>
+```
+
+Then say you wrote this CSS for it:
+
+```css
+p {
+  width: 100px;
+  height: 50px;
+  padding: 20px;
+  border: 1px solid;
+}
+```
+
+The content would break out of your element and it would be 140px
+wide, rather than 100px. Why is that? The box model is a core
+foundation of CSS and understanding how it works, how it is affected
+by other aspects of CSS and importantly, how you can control it will
+help you to write more predictable CSS.
+
+<iframe height="500" style="width: 100%;" scrolling="no" title="Learn CSS - Box model intro" src="https://codepen.io/piccalilli/embed/0c03f5dc8a83ccaa2f48fcc115e1ec54?height=500&theme-id=38982&default-tab=result" frameborder="no" loading="lazy" allowtransparency="true" allowfullscreen="true">
+  See the Pen <a href='https://codepen.io/piccalilli/pen/0c03f5dc8a83ccaa2f48fcc115e1ec54'>Learn CSS - Box model intro</a> by Piccalilli
+  (<a href='https://codepen.io/piccalilli'>@piccalilli</a>) on <a href='https://codepen.io'>CodePen</a>.
+</iframe>
+
+---
+
+A really important thing to remember when writing CSS—or working on
+the web as a whole—is that everything in the browser is a box. Whether
+that's a box that uses `border-radius` to look like a circle, or even
+just some text: the key thing to remember is that it's all boxes.
+
+## Content and sizing
+
+Boxes have different behaviour based on their `display` value, their
+set dimensions, and the content that lives within them. This content
+could be even more boxes—like child elements—or plain text content.
+Either way, **this content will affect the size of the box by
+default**.
+
+You can control this by using **extrinsic sizing**, or, you can
+continue to let the browser make decisions for us, using **intrinsic
+sizing**.
+
+Let's quickly look at the difference, using a demo to help us.
+
+<iframe height="500" style="width: 100%;" scrolling="no" title="Learn CSS - Box model intro" src="https://codepen.io/piccalilli/embed/5cbda8499e19245d7b48cac8d796dda5?height=500&theme-id=38982&default-tab=result" frameborder="no" loading="lazy" allowtransparency="true" allowfullscreen="true">
+  See the Pen <a href='https://codepen.io/piccalilli/pen/5cbda8499e19245d7b48cac8d796dda5'>Learn CSS - Box model intro</a> by Piccalilli
+  (<a href='https://codepen.io/piccalilli'>@piccalilli</a>) on <a href='https://codepen.io'>CodePen</a>.
+</iframe>
+
+[CAPTION: Notice that when the box is using **extrinsic sizing**,
```

Comment:
I noticed this is also inside of the codepen. Did you want to leave it in the codepen? If you'd prefer to have it on the page I can look to see if we have an existing caption style.

Reply:
I'm torn with these at the moment tbh. Later demos tend to be standalone. We could implement a caption parameter to the CodePen shortcode maybe? Then if that's not an empty string, render out the embed and caption as a `<figure>` and `<figcaption>`?